### PR TITLE
feat: extend go logger rule

### PR DESCRIPTION
**Pull request by elsapet**
## Description

Add more logger methods to existing go_lang_logger rule
Add new go_lang_logger_leak rule (any non-string input)

## Checklist

- [ ] I've added a snapshot that shows my rule works as expected.
- [ ] My rule has adequate metadata to explain its use.
- [ ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

**Review by cfabianski — APPROVED** [review]

File: tests/go/lang/logger_leak/testdata/main.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	zerolog "github.com/rs/zerolog"
+	logrus "github.com/sirupsen/logrus"
+	"go.uber.org/zap"
+)
+
+func bad(Data any) {
+	// bearer:expected go_lang_logger_leak
+	zerolog.Error().Msg(Data)
+	// bearer:expected go_lang_logger_leak
+	zerolog.Error().Msgf("Some data: %#v", Data)
```
Comment (elsapet) [inline tests/go/lang/logger_leak/testdata/main.go:15]: These are also counted as string literals, as in the Java case

Comment (elsapet) [reply tests/go/lang/logger_leak/testdata/main.go:15]: https://github.com/Bearer/bearer-rules/actions/runs/8001834206/job/21853837960?pr=301

File: rules/go/lang/logger_leak.yml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+          - variable: CALLER
+            detection: go_lang_logger_leak_logger
+          - variable: CALLER
+            detection: go_lang_logger_leak_zerolog
+      - variable: METHOD
+        regex: \A(Fatal|Panic|Print)(f|ln)?\z
+      - not:
+          variable: INPUT
+          detection: string_literal
+          scope: result
```
Comment (cfabianski) [inline rules/go/lang/logger_leak.yml:17]:
```suggestion
          scope: cursor
```

File: rules/go/lang/logger_leak.yml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      - variable: EVENT
+        regex: \A(Info|Debug|Error|Trace|Fatal|Panic|Warn)\z
+      - variable: METHOD
+        values:
+          - Msg
+          - Msgf
+      - not:
+          variable: INPUT
+          detection: string_literal
+          scope: result
```
Comment (cfabianski) [inline rules/go/lang/logger_leak.yml:39]:
```suggestion
          scope: cursor
```

File: rules/go/lang/logger_leak.yml [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+          - variable: CALLER
+            detection: go_lang_logger_leak_logrus
+          - variable: CALLER
+            detection: go_lang_logger_leak_seelog
+      - variable: METHOD
+        regex: \A(WithFields\.)?(Info|Debug|Error|Trace|Fatal|Panic|Warn)\z
+      - not:
+          variable: INPUT
+          detection: string_literal
+          scope: result
```
Comment (cfabianski) [inline rules/go/lang/logger_leak.yml:57]:
```suggestion
          scope: cursor
```

File: rules/go/lang/logger_leak.yml [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+      $<CALLER>.$<METHOD>($<INPUT>);
+    filters:
+      - variable: CALLER
+        detection: go_lang_logger_leak_glog
+      - variable: METHOD
+        regex: \A(Info|Warning|Error|Fatal)(Contex)?(Depth)?(f)?\z
+      - not:
+          variable: INPUT
+          detection: string_literal
+          scope: result
```
Comment (cfabianski) [inline rules/go/lang/logger_leak.yml:72]:
```suggestion
          scope: cursor
```

File: rules/go/lang/logger_leak.yml [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+      $<ZAP>.$<METHOD>($<INPUT>);
+    filters:
+      - variable: ZAP
+        detection: go_lang_logger_leak_zap
+      - variable: METHOD
+        regex: \A(WithFields\.)?(Info|Log|Error|Fatal|DPanic|Warn)\z
+      - not:
+          variable: INPUT
+          detection: string_literal
+          scope: result
```
Comment (cfabianski) [inline rules/go/lang/logger_leak.yml:87]:
```suggestion
          scope: cursor
```

**Pull request merged**